### PR TITLE
Added tests for an Actions

### PR DESCRIPTION
**Pull request by krolwofficial**
Tests for an Actions component

### CODE REVIEW

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  };
+});
+
+jest.mock('../../Group', () => {
+  return  () => {
+    return <div className="Group" />;
+  };
+});
+
+describe('<Actions />', () => {
```
Comment (djdeimon) [inline src/alto-ui/Actions/__tests__/Actions.test.js]:
```suggestion
describe('Actions', () => {
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Actions from '../Actions';
+
```
Comment (djdeimon) [inline src/alto-ui/Actions/__tests__/Actions.test.js]: Remove this empty line
```suggestion
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    const wrapper = getWrapper({items: [{key: '1'}, {key: '2'}], max: 1 });
+    const DropdownMock = wrapper.find(Dropdown);
+
+    expect(DropdownMock).toHaveLength(1);
+  });
+
+    it('should return undefined if onClick props is not a function', () => {
+      const wrapper = getWrapper({});
+      const DropdownMock = wrapper.find(Dropdown);
+      expect(DropdownMock.props().onClick).toBeUndefined();
```
Comment (krlakwa) [inline src/alto-ui/Actions/__tests__/Actions.test.js]: try like this:
```suggestion
      expect(DropdownMock.prop('onClick')).toBeUndefined();
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          title: 'Edit',
+        },
+      ],
+      max: 1
+    };
+
+  const getWrapper = ({ ...props }) => shallow(<Actions {...defaultProps} {...props} />);
+
+  it('should not render if no items are provided', () => {
+    expect(getWrapper({items: undefined}).type()).toBe(null);
```
Comment (krlakwa) [inline src/alto-ui/Actions/__tests__/Actions.test.js]:
```suggestion
    expect(getWrapper({items: undefined}).type()).toBeNull();
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+describe('<Actions />', () => {
+    const defaultProps = {
+      className: 'classTest',
+      id: 'id',
+      items: [
+        {
+          icon: null,
+          key: '1',
+          onClick: () => [],
```
Comment (krlakwa) [inline src/alto-ui/Actions/__tests__/Actions.test.js]: I think you should use jest.mock()
```suggestion
          onClick: jest.mock(),
```

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 1 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: fixed in "fixes after code review" commit

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 2 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: fixed in "fixes after code review" commit

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 3 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: This test as we talked was removed

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 4 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: fixed in "fixes after code review" commit

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 5 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: fixed in "fixes after code review" commit

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import Actions from '../Actions';
+import Dropdown from '../../Dropdown';
+import Group from '../../Group';
+
+jest.mock('../../Dropdown', () => {
```
Comment (barteksiudak) [inline src/alto-ui/Actions/__tests__/Actions.test.js]: `jest.mock('../../Dropdown', () => ({ onClick }) => <button className="Dropdown" onClick={() => onClick()} />;`

**Comment by barteksiudak** [comment]
There are missing some tests:
1. <Dropdown /> `selected` prop should be checked
2. <Dropdown /> `items` prop should be checked ( and separated test for `handleClick` )

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 6 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: 1, Missing tests are added
2. The Dropdown mock has been rebuilt

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        icon: null,
+        key: '2',
+        onClick: jest.fn(),
+        title: 'Edit',
+      },
+    ],
+    max: 1,
+  };
+
+  const getWrapper = ({ ...props }) => mount(<Actions {...defaultProps} {...props} />);
```
Comment (djdeimon) [inline src/alto-ui/Actions/__tests__/Actions.test.js]:
```suggestion
  const getWrapper = props  => mount(<Actions {...defaultProps} {...props} />);
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        {
+          key: '2',
+          title: 'Edit',
+        },
+      ],
+      max: 3,
+      onClick: mockOnClick(),
+    };
+
+    const wrapper = getWrapper({ props });
```
Comment (djdeimon) [inline src/alto-ui/Actions/__tests__/Actions.test.js]:
```suggestion
    const wrapper = getWrapper( props );
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+          key: '1',
+          title: 'Edit',
+        },
+        {
+          key: '2',
+          title: 'Edit',
+        },
+      ],
+      max: 3,
+      onClick: mockOnClick(),
```
Comment (djdeimon) [inline src/alto-ui/Actions/__tests__/Actions.test.js]:
```suggestion
      onClick: mockOnClick,
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+import { mount } from 'enzyme';
+
+import Actions from '../Actions';
+import Dropdown from '../../Dropdown';
+import Group from '../../Group';
+
+jest.mock('../../Dropdown', () => {
+  return props => {
+    return (
+      <div className="Dropdown" onClick={() => props.onClick} onKeyDown={() => props.onClick} role="menu" tabIndex="0">
```
Comment (djdeimon) [inline src/alto-ui/Actions/__tests__/Actions.test.js]:
```suggestion
      <div className="Dropdown" onClick={props.onClick} onKeyDown={props.onClick} role="menu" tabIndex="0">
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+        {props.items.map(action => {
+          return (
+            <button key={`key_${action.key}`} id={`key_${action.key}`} onClick={() => action.onClick || props.onClick()} />
```
Comment (djdeimon) [inline src/alto-ui/Actions/__tests__/Actions.test.js]:
```suggestion
            <button key={`key_${action.key}`} id={`key_${action.key}`} onClick={action.onClick || props.onClick} />
```

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 7 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Refactored in "Refactor after code review" commit

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 8 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Refactored in "Refactor after code review" commit

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 11 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Refactored in "Refactor after code review" commit

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+            {key: '1', title: '1'},
+          ],
+          onClick: mockedHandleClick,
+        }
+
+        wrapper = getWrapper(defaultProps, props);
+        const button = wrapper.find('button')
+        button.simulate('click')
+
+        expect(mockedHandleClick).toHaveBeenCalledTimes(1)
```
Comment (barteksiudak) [inline src/alto-ui/Actions/__tests__/Actions.test.js]: Check if arguments for called function are correct `toHaveBeenCalledWith(...)`

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+          items: [
+            {key: '1', title: '1', onClick: mockedOnClick},
+          ],
+        }
+
+        wrapper = getWrapper(defaultProps, props);
+        const button = wrapper.find('button')
+        button.simulate('click')
+
+        expect(mockedOnClick).toHaveBeenCalledTimes(1)
```
Comment (barteksiudak) [inline src/alto-ui/Actions/__tests__/Actions.test.js]: Check also `toHaveBeenCalledWith(...)`

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+
+jest.mock('../../Group', () => {
+  return props => {
+    return (
+      <div role="menu" tabIndex="0" items={props.items} />
+    )
+  };
+});
+
+describe('Actions', () => {
```
Comment (barteksiudak) [inline src/alto-ui/Actions/__tests__/Actions.test.js]: I want a test that `Actions` returns `null` if no items.

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 12 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Added `expect(mockedHandleClick).toHaveBeenCalledWith(obj)`

File: src/alto-ui/Actions/__tests__/Actions.test.js (hunk 14 above)
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Added `it('returns null if no items provided', () =>` test

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+  });
+
+  describe('should return Dropdown component', () => {
+    let wrapper;
+    let defaultProps;
+    let items;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      items = [
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Refactored 
```
const defaultItems = [
    { key: '1', title: '1', active: true },
    { key: '2', title: '2'},
    { key: '3', title: '3', active: true },
  ];
```

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 16]
```diff
@@ -0,0 +1,141 @@
...
+              onClick={action.onClick || props.onClick}
+            />
+          );
+        })}
+      </div>
+    );
+  };
+});
+
+jest.mock('../../Group', () => {
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Refactored

File: src/alto-ui/Actions/__tests__/Actions.test.js [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+
+import Actions from '../Actions';
+import Dropdown from '../../Dropdown';
+import Group from '../../Group';
+
+jest.mock('../../Dropdown', () => {
+  return props => {
+    return (
+      <div role="menu" tabIndex="0">
+        {props.items.map(action => {
```
[the remark this answers is not in the document]
Comment (krolwofficial) [reply src/alto-ui/Actions/__tests__/Actions.test.js]: Refactored

**Pull request merged**